// @tinkermonkey/heimdall-ui/dist/index.js
import b, { useState as K, useRef as W, useEffect as Y, useImperativeHandle as _e } from "react";
var me = { exports: {} }, J = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ye;
function Ma() {
  if (Ye) return J;
  Ye = 1;
  var r = b, t = Symbol.for("react.element"), s = Symbol.for("react.fragment"), i = Object.prototype.hasOwnProperty, o = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, l = { key: !0, ref: !0, __self: !0, __source: !0 };
  function d(m, u, p) {
    var C, v = {}, j = null, f = null;
    p !== void 0 && (j = "" + p), u.key !== void 0 && (j = "" + u.key), u.ref !== void 0 && (f = u.ref);
    for (C in u) i.call(u, C) && !l.hasOwnProperty(C) && (v[C] = u[C]);
    if (m && m.defaultProps) for (C in u = m.defaultProps, u) v[C] === void 0 && (v[C] = u[C]);
    return { $$typeof: t, type: m, key: j, ref: f, props: v, _owner: o.current };
  }
  return J.Fragment = s, J.jsx = d, J.jsxs = d, J;
}
var G = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ue;
function Ea() {
  return Ue || (Ue = 1, process.env.NODE_ENV !== "production" && function() {
    var r = b, t = Symbol.for("react.element"), s = Symbol.for("react.portal"), i = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), l = Symbol.for("react.profiler"), d = Symbol.for("react.provider"), m = Symbol.for("react.context"), u = Symbol.for("react.forward_ref"), p = Symbol.for("react.suspense"), C = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), j = Symbol.for("react.lazy"), f = Symbol.for("react.offscreen"), y = Symbol.iterator, k = "@@iterator";
    function P(e) {
      if (e === null || typeof e != "object")
        return null;
      var n = y && e[y] || e[k];
      return typeof n == "function" ? n : null;
    }
    var M = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function E(e) {
      {
        for (var n = arguments.length, c = new Array(n > 1 ? n - 1 : 0), h = 1; h < n; h++)
          c[h - 1] = arguments[h];
        L("error", e, c);
      }
    }
    function L(e, n, c) {
      {
        var h = M.ReactDebugCurrentFrame, N = h.getStackAddendum();
        N !== "" && (n += "%s", c = c.concat([N]));
        var g = c.map(function(x) {
          return String(x);
        });
        g.unshift("Warning: " + n), Function.prototype.apply.call(console[e], console, g);
      }
    }
    var S = !1, O = !1, A = !1, $ = !1, ne = !1, be;
    be = Symbol.for("react.module.reference");
    function Ge(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === i || e === l || ne || e === o || e === p || e === C || $ || e === f || S || O || A || typeof e == "object" && e !== null && (e.$$typeof === j || e.$$typeof === v || e.$$typeof === d || e.$$typeof === m || e.$$typeof === u || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === be || e.getModuleId !== void 0));
    }
    function Xe(e, n, c) {
      var h = e.displayName;
      if (h)
        return h;
      var N = n.displayName || n.name || "";
      return N !== "" ? c + "(" + N + ")" : c;
    }
    function Le(e) {
      return e.displayName || "Context";
    }
    function F(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && E("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case i:
          return "Fragment";
        case s:
          return "Portal";
        case l:
          return "Profiler";
        case o:
          return "StrictMode";
        case p:
          return "Suspense";
        case C:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case m:
            var n = e;
            return Le(n) + ".Consumer";
          case d:
            var c = e;
            return Le(c._context) + ".Provider";
          case u:
            return Xe(e, e.render, "ForwardRef");
          case v:
            var h = e.displayName || null;
            return h !== null ? h : F(e.type) || "Memo";
          case j: {
            var N = e, g = N._payload, x = N._init;
            try {
              return F(x(g));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var I = Object.assign, H = 0, xe, je, ye, Ne, ge, Me, Ee;
    function Re() {
    }
    Re.__reactDisabledLog = !0;
    function Qe() {
      {
        if (H === 0) {
          xe = console.log, je = console.info, ye = console.warn, Ne = console.error, ge = console.group, Me = console.groupCollapsed, Ee = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: Re,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        H++;
      }
    }
    function ea() {
      {
        if (H--, H === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: I({}, e, {
              value: xe
            }),
            info: I({}, e, {
              value: je
            }),
            warn: I({}, e, {
              value: ye
            }),
            error: I({}, e, {
              value: Ne
            }),
            group: I({}, e, {
              value: ge
            }),
            groupCollapsed: I({}, e, {
              value: Me
            }),
            groupEnd: I({}, e, {
              value: Ee
            })
          });
        }
        H < 0 && E("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var se = M.ReactCurrentDispatcher, ie;
    function X(e, n, c) {
      {
        if (ie === void 0)
          try {
            throw Error();
          } catch (N) {
            var h = N.stack.trim().match(/\n( *(at )?)/);
            ie = h && h[1] || "";
          }
        return `
` + ie + e;
      }
    }
    var le = !1, Q;
    {
      var aa = typeof WeakMap == "function" ? WeakMap : Map;
      Q = new aa();
    }
    function we(e, n) {
      if (!e || le)
        return "";
      {
        var c = Q.get(e);
        if (c !== void 0)
          return c;
      }
      var h;
      le = !0;
      var N = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var g;
      g = se.current, se.current = null, Qe();
      try {
        if (n) {
          var x = function() {
            throw Error();
          };
          if (Object.defineProperty(x.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(x, []);
            } catch (D) {
              h = D;
            }
            Reflect.construct(e, [], x);
          } else {
            try {
              x.call();
            } catch (D) {
              h = D;
            }
            e.call(x.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (D) {
            h = D;
          }
          e();
        }
      } catch (D) {
        if (D && h && typeof D.stack == "string") {
          for (var _ = D.stack.split(`
`), T = h.stack.split(`
`), R = _.length - 1, w = T.length - 1; R >= 1 && w >= 0 && _[R] !== T[w]; )
            w--;
          for (; R >= 1 && w >= 0; R--, w--)
            if (_[R] !== T[w]) {
              if (R !== 1 || w !== 1)
                do
                  if (R--, w--, w < 0 || _[R] !== T[w]) {
                    var Z = `
` + _[R].replace(" at new ", " at ");
                    return e.displayName && Z.includes("<anonymous>") && (Z = Z.replace("<anonymous>", e.displayName)), typeof e == "function" && Q.set(e, Z), Z;
                  }
                while (R >= 1 && w >= 0);
              break;
            }
        }
      } finally {
        le = !1, se.current = g, ea(), Error.prepareStackTrace = N;
      }
      var V = e ? e.displayName || e.name : "", z = V ? X(V) : "";
      return typeof e == "function" && Q.set(e, z), z;
    }
    function ra(e, n, c) {
      return we(e, !1);
    }
    function ta(e) {
      var n = e.prototype;
      return !!(n && n.isReactComponent);
    }
    function ee(e, n, c) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return we(e, ta(e));
      if (typeof e == "string")
        return X(e);
      switch (e) {
        case p:
          return X("Suspense");
        case C:
          return X("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case u:
            return ra(e.render);
          case v:
            return ee(e.type, n, c);
          case j: {
            var h = e, N = h._payload, g = h._init;
            try {
              return ee(g(N), n, c);
            } catch {
            }
          }
        }
      return "";
    }
    var q = Object.prototype.hasOwnProperty, ke = {}, Te = M.ReactDebugCurrentFrame;
    function ae(e) {
      if (e) {
        var n = e._owner, c = ee(e.type, e._source, n ? n.type : null);
        Te.setExtraStackFrame(c);
      } else
        Te.setExtraStackFrame(null);
    }
    function na(e, n, c, h, N) {
      {
        var g = Function.call.bind(q);
        for (var x in e)
          if (g(e, x)) {
            var _ = void 0;
            try {
              if (typeof e[x] != "function") {
                var T = Error((h || "React class") + ": " + c + " type `" + x + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[x] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw T.name = "Invariant Violation", T;
              }
              _ = e[x](n, x, h, c, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (R) {
              _ = R;
            }
            _ && !(_ instanceof Error) && (ae(N), E("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", h || "React class", c, x, typeof _), ae(null)), _ instanceof Error && !(_.message in ke) && (ke[_.message] = !0, ae(N), E("Failed %s type: %s", c, _.message), ae(null));
          }
      }
    }
    var sa = Array.isArray;
    function oe(e) {
      return sa(e);
    }
    function ia(e) {
      {
        var n = typeof Symbol == "function" && Symbol.toStringTag, c = n && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return c;
      }
    }
    function la(e) {
      try {
        return Pe(e), !1;
      } catch {
        return !0;
      }
    }
    function Pe(e) {
      return "" + e;
    }
    function Se(e) {
      if (la(e))
        return E("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ia(e)), Pe(e);
    }
    var De = M.ReactCurrentOwner, oa = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Oe, Ze;
    function ca(e) {
      if (q.call(e, "ref")) {
        var n = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (n && n.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function da(e) {
      if (q.call(e, "key")) {
        var n = Object.getOwnPropertyDescriptor(e, "key").get;
        if (n && n.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function ua(e, n) {
      typeof e.ref == "string" && De.current;
    }
    function fa(e, n) {
      {
        var c = function() {
          Oe || (Oe = !0, E("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", n));
        };
        c.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: c,
          configurable: !0
        });
      }
    }
    function ma(e, n) {
      {
        var c = function() {
          Ze || (Ze = !0, E("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", n));
        };
        c.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: c,
          configurable: !0
        });
      }
    }
    var ha = function(e, n, c, h, N, g, x) {
      var _ = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: e,
        key: n,
        ref: c,
        props: x,
        // Record the component responsible for creating this element.
        _owner: g
      };
      return _._store = {}, Object.defineProperty(_._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(_, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: h
      }), Object.defineProperty(_, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: N
      }), Object.freeze && (Object.freeze(_.props), Object.freeze(_)), _;
    };
    function va(e, n, c, h, N) {
      {
        var g, x = {}, _ = null, T = null;
        c !== void 0 && (Se(c), _ = "" + c), da(n) && (Se(n.key), _ = "" + n.key), ca(n) && (T = n.ref, ua(n, N));
        for (g in n)
          q.call(n, g) && !oa.hasOwnProperty(g) && (x[g] = n[g]);
        if (e && e.defaultProps) {
          var R = e.defaultProps;
          for (g in R)
            x[g] === void 0 && (x[g] = R[g]);
        }
        if (_ || T) {
          var w = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          _ && fa(x, w), T && ma(x, w);
        }
        return ha(e, _, T, N, h, De.current, x);
      }
    }
    var ce = M.ReactCurrentOwner, Be = M.ReactDebugCurrentFrame;
    function U(e) {
      if (e) {
        var n = e._owner, c = ee(e.type, e._source, n ? n.type : null);
        Be.setExtraStackFrame(c);
      } else
        Be.setExtraStackFrame(null);
    }
    var de;
    de = !1;
    function ue(e) {
      return typeof e == "object" && e !== null && e.$$typeof === t;
    }
    function $e() {
      {
        if (ce.current) {
          var e = F(ce.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function _a(e) {
      return "";
    }
    var Ae = {};
    function pa(e) {
      {
        var n = $e();
        if (!n) {
          var c = typeof e == "string" ? e : e.displayName || e.name;
          c && (n = `

Check the top-level render call using <` + c + ">.");
        }
        return n;
      }
    }
    function Fe(e, n) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var c = pa(n);
        if (Ae[c])
          return;
        Ae[c] = !0;
        var h = "";
        e && e._owner && e._owner !== ce.current && (h = " It was passed a child from " + F(e._owner.type) + "."), U(e), E('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', c, h), U(null);
      }
    }
    function Ie(e, n) {
      {
        if (typeof e != "object")
          return;
        if (oe(e))
          for (var c = 0; c < e.length; c++) {
            var h = e[c];
            ue(h) && Fe(h, n);
          }
        else if (ue(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var N = P(e);
          if (typeof N == "function" && N !== e.entries)
            for (var g = N.call(e), x; !(x = g.next()).done; )
              ue(x.value) && Fe(x.value, n);
        }
      }
    }
    function Ca(e) {
      {
        var n = e.type;
        if (n == null || typeof n == "string")
          return;
        var c;
        if (typeof n == "function")
          c = n.propTypes;
        else if (typeof n == "object" && (n.$$typeof === u || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        n.$$typeof === v))
          c = n.propTypes;
        else
          return;
        if (c) {
          var h = F(n);
          na(c, e.props, "prop", h, e);
        } else if (n.PropTypes !== void 0 && !de) {
          de = !0;
          var N = F(n);
          E("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", N || "Unknown");
        }
        typeof n.getDefaultProps == "function" && !n.getDefaultProps.isReactClassApproved && E("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ba(e) {
      {
        for (var n = Object.keys(e.props), c = 0; c < n.length; c++) {
          var h = n[c];
          if (h !== "children" && h !== "key") {
            U(e), E("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", h), U(null);
            break;
          }
        }
        e.ref !== null && (U(e), E("Invalid attribute `ref` supplied to `React.Fragment`."), U(null));
      }
    }
    var ze = {};
    function We(e, n, c, h, N, g) {
      {
        var x = Ge(e);
        if (!x) {
          var _ = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (_ += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var T = _a();
          T ? _ += T : _ += $e();
          var R;
          e === null ? R = "null" : oe(e) ? R = "array" : e !== void 0 && e.$$typeof === t ? (R = "<" + (F(e.type) || "Unknown") + " />", _ = " Did you accidentally export a JSX literal instead of a component?") : R = typeof e, E("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", R, _);
        }
        var w = va(e, n, c, N, g);
        if (w == null)
          return w;
        if (x) {
          var Z = n.children;
          if (Z !== void 0)
            if (h)
              if (oe(Z)) {
                for (var V = 0; V < Z.length; V++)
                  Ie(Z[V], e);
                Object.freeze && Object.freeze(Z);
              } else
                E("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ie(Z, e);
        }
        if (q.call(n, "key")) {
          var z = F(e), D = Object.keys(n).filter(function(ga) {
            return ga !== "key";
          }), fe = D.length > 0 ? "{key: someKey, " + D.join(": ..., ") + ": ...}" : "{key: someKey}";
          if (!ze[z + fe]) {
            var Na = D.length > 0 ? "{" + D.join(": ..., ") + ": ...}" : "{}";
            E(`A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`, fe, z, Na, z), ze[z + fe] = !0;
          }
        }
        return e === i ? ba(w) : Ca(w), w;
      }
    }
    function La(e, n, c) {
      return We(e, n, c, !0);
    }
    function xa(e, n, c) {
      return We(e, n, c, !1);
    }
    var ja = xa, ya = La;
    G.Fragment = i, G.jsx = ja, G.jsxs = ya;
  }()), G;
}
process.env.NODE_ENV === "production" ? me.exports = Ma() : me.exports = Ea();
var a = me.exports;
const Ra = {
  dashboard: "M3 3 L10 3 L10 10 L3 10 Z M14 3 L21 3 L21 10 L14 10 Z M14 14 L21 14 L21 21 L14 21 Z M3 14 L10 14 L10 21 L3 21 Z",
  schema: "M3 12 L9 6 L21 6 M3 12 L9 18 L21 18 M3 12 C4.7 12 6 10.3 6 8 C6 5.7 4.7 4 3 4 C1.3 4 0 5.7 0 8 C0 10.3 1.3 12 3 12 M21 6 C22.7 6 24 4.3 24 2 C24 0.3 22.7 0 21 0 C19.3 0 18 0.3 18 2 C18 4.3 19.3 6 21 6 M21 18 C22.7 18 24 16.3 24 14 C24 12.3 22.7 11 21 11 C19.3 11 18 12.3 18 14 C18 16.3 19.3 18 21 18",
  data: "M12 5 C18.6 5 24 6.3 24 8 L24 8 L24 16 C24 17.7 18.6 19 12 19 C5.4 19 0 17.7 0 16 L0 8 C0 6.3 5.4 5 12 5 M3 5 L3 14 L21 14 L21 5 M3 11 L3 19 L21 19 L21 11",
  pipeline: "M2 12 L6 12 L6 6 L18 6 L18 12 L22 12 M8 12 L16 12 M12 9 L15 12 L12 15",
  graph: "M6 10 C8.8 10 11 7.8 11 5 C11 2.2 8.8 0 6 0 C3.2 0 1 2.2 1 5 C1 7.8 3.2 10 6 10 M18 10 C20.8 10 23 7.8 23 5 C23 2.2 20.8 0 18 0 C15.2 0 13 2.2 13 5 C13 7.8 15.2 10 18 10 M12 18 C14.8 18 17 15.8 17 13 C17 10.2 14.8 8 12 8 C9.2 8 7 10.2 7 13 C7 15.8 9.2 18 12 18 M8 8 L16 16 M16 8 L8 16 M12 9 L12 15",
  search: "M10.6 14.8 C14.2 14.8 17.2 11.5 17.2 7.4 C17.2 3 14.2 0 10.6 0 C6.9 0 3 3 3 7.4 C3 11.5 6.9 14.8 10.6 14.8 M19.8 19.2 L16.7 16.3",
  bell: "M6.55 9.7 C6.55 3.7 11.15 0 11.15 0 C11.15 0 15.75 3.7 15.75 9.7 C15.75 12 17.25 13.4 17.25 13.4 L4.6 13.4 C4.6 13.4 6.55 12 6.55 9.7 M9.4 19.4 C9.73 19.8 10.8 20.1 10.95 20.1 C11.1 20.1 12.18 19.8 12.5 19.4",
  plus: "M12 5 L12 19 M5 12 L19 12",
  check: "M2 12 L8 18 L22 4",
  x: "M3 3 L21 21 M21 3 L3 21",
  chevronDown: "M6 9 L12 15 L18 9",
  chevronUp: "M18 15 L12 9 L6 15",
  chevronLeft: "M15 6 L9 12 L15 18",
  chevronRight: "M9 6 L15 12 L9 18",
  menu: "M3 6 L21 6 M3 12 L21 12 M3 18 L21 18",
  settings: "M12 1.5 C12.4 1.5 12.7 1.8 12.7 2.2 L12.7 4 C14.2 4.4 15.6 5.3 16.7 6.6 L18 5.4 C18.4 5.1 19 5.1 19.3 5.4 C19.7 5.7 19.7 6.3 19.3 6.6 L18 7.8 C19.2 8.9 20.2 10.3 20.6 12 L22.4 12 C22.9 12 23.2 12.3 23.2 12.8 C23.2 13.3 22.9 13.6 22.4 13.6 L20.6 13.6 C20.2 15.3 19.2 16.7 18 17.8 L19.3 19 C19.7 19.3 19.7 19.9 19.3 20.2 C19 20.5 18.4 20.5 18 20.2 L16.7 19 C15.6 20.3 14.2 21.2 12.7 21.6 L12.7 23.4 C12.7 23.8 12.4 24.1 12 24.1 C11.6 24.1 11.3 23.8 11.3 23.4 L11.3 21.6 C9.8 21.2 8.4 20.3 7.3 19 L6 20.2 C5.6 20.5 5 20.5 4.7 20.2 C4.3 19.9 4.3 19.3 4.7 19 L6 17.8 C4.8 16.7 3.8 15.3 3.4 13.6 L1.6 13.6 C1.1 13.6 0.8 13.3 0.8 12.8 C0.8 12.3 1.1 12 1.6 12 L3.4 12 C3.8 10.3 4.8 8.9 6 7.8 L4.7 6.6 C4.3 6.3 4.3 5.7 4.7 5.4 C5 5.1 5.6 5.1 6 5.4 L7.3 6.6 C8.4 5.3 9.8 4.4 11.3 4 L11.3 2.2 C11.3 1.8 11.6 1.5 12 1.5 Z M12 8.5 C10.6 8.5 9.5 9.6 9.5 11 C9.5 12.4 10.6 13.5 12 13.5 C13.4 13.5 14.5 12.4 14.5 11 C14.5 9.6 13.4 8.5 12 8.5 Z",
  alert: "M12 2 L2 20 L22 20 Z M12 9 L12 13 M12 17 L12.01 17",
  trash: "M3 6 L3 20 C3 21.1 3.9 22 5 22 L19 22 C20.1 22 21 21.1 21 20 L21 6 M1 6 L23 6 M7 6 L7 4 C7 2.9 7.9 2 9 2 L15 2 C16.1 2 17 2.9 17 4 L17 6 M9 11 L9 17 M15 11 L15 17",
  edit: "M3 17.25 L3 21 L6.75 21 L17.81 9.94 L14.06 6.19 Z M20.71 7.04 C21.1 6.65 21.1 6.02 20.71 5.63 L18.37 3.29 C17.98 2.9 17.35 2.9 16.96 3.29 L15.12 5.12 L18.87 8.87 Z",
  download: "M12 2 L12 13 M5 10 L12 17 L19 10",
  upload: "M12 22 L12 11 M5 14 L12 7 L19 14",
  eye: "M12 5 C7 5 2.73 8.11 1 12.46 C2.73 16.89 7 20 12 20 C17 20 21.27 16.89 23 12.46 C21.27 8.11 17 5 12 5 Z M12 16 C14.2 16 16 14.2 16 12 C16 9.8 14.2 8 12 8 C9.8 8 8 9.8 8 12 C8 14.2 9.8 16 12 16 Z",
  eyeOff: "M12 5.69 C8.5 5.69 5.2 7.6 3.2 10.4 M21 19 L2 2 M9.88 9.88 C9.4 10.5 9 11.2 9 12 C9 14.21 10.79 16 13 16 C13.8 16 14.5 15.6 15.12 15.12 M18.8 10.4 C20.8 7.6 17.5 5.69 12 5.69",
  clock: "M12 2 C6.48 2 2 6.48 2 12 C2 17.52 6.48 22 12 22 C17.52 22 22 17.52 22 12 C22 6.48 17.52 2 12 2 Z M12 20 C7.58 20 4 16.42 4 12 C4 7.58 7.58 4 12 4 C16.42 4 20 7.58 20 12 C20 16.42 16.42 20 12 20 Z M13 7 L11 7 L11 13 L16.25 15.85 L17 14.32 L13 12 Z",
  calendar: "M5 2 L5 4 L19 4 L19 2 M3 4 L3 20 C3 21.1 3.9 22 5 22 L19 22 C20.1 22 21 21.1 21 20 L21 4 M7 11 L7 13 L9 13 L9 11 L7 11 Z M12 11 L12 13 L14 13 L14 11 L12 11 Z M17 11 L17 13 L19 13 L19 11 L17 11 Z",
  filter: "M3 6 L21 6 L18 13 L18 20 L6 20 L6 13 Z",
  link: "M10 11 C10.8 10.3 12 10.2 12.7 10.2 L20 10.2 C21 10.2 21.8 11 21.8 12 L21.8 19 C21.8 20 21 20.8 20 20.8 L12.7 20.8 C11.7 20.8 10.9 20 10.9 19 L10.9 12 C10.9 11.9 10.91 11.8 10.92 11.7 M7 11 C7.8 10.3 6.5 10.2 5.8 10.2 L1 10.2 C0.1 10.2 0.2 11 0.2 12 L0.2 19 C0.2 20 0.9 20.8 1.8 20.8 L5.8 20.8 C6.8 20.8 7.6 20 7.6 19 L7.6 12 C7.6 11.9 7.59 11.8 7.58 11.7",
  lock: "M7 10 L7 7 C7 4.24 9.24 2 12 2 C14.76 2 17 4.24 17 7 L17 10 M5 10 L19 10 L19 20 C19 21.1 18.1 22 17 22 L7 22 C5.9 22 5 21.1 5 20 Z",
  unlock: "M7 10 L19 10 L19 20 C19 21.1 18.1 22 17 22 L7 22 C5.9 22 5 21.1 5 20 L5 10 L5 7 C5 4.24 7.24 2 10 2 C12.76 2 15 4.24 15 7 L13 7 C13 5.34 11.66 4 10 4 C8.34 4 7 5.34 7 7 Z",
  user: "M12 12 C13.66 12 15 10.66 15 9 C15 7.34 13.66 6 12 6 C10.34 6 9 7.34 9 9 C9 10.66 10.34 12 12 12 Z M12 22 C16.42 22 21 20 21 16 L21 14 C21 11.79 19.21 10 17 10 L7 10 C4.79 10 3 11.79 3 14 L3 16 C3 20 7.58 22 12 22 Z",
  copy: "M16 1 L4 1 C2.9 1 2 1.9 2 3 L2 17 L4 17 L4 3 L16 3 Z M20 5 L8 5 C6.9 5 6 5.9 6 7 L6 21 C6 22.1 6.9 23 8 23 L20 23 C21.1 23 22 22.1 22 21 L22 7 C22 5.9 21.1 5 20 5 Z",
  info: "M12 2 C6.48 2 2 6.48 2 12 C2 17.52 6.48 22 12 22 C17.52 22 22 17.52 22 12 C22 6.48 17.52 2 12 2 Z M13 17 L11 17 L11 11 L13 11 Z M13 9 L11 9 L11 7 L13 7 Z",
  help: "M12 2 C6.48 2 2 6.48 2 12 C2 17.52 6.48 22 12 22 C17.52 22 22 17.52 22 12 C22 6.48 17.52 2 12 2 Z M13 19 L11 19 L11 17 L13 17 Z M15.07 10.25 C14.29 11.29 13.5 12 13 12.9 L13 14 L11 14 L11 13 C11 11.52 11.79 10.29 12.93 9.07 C13.62 8.29 14.29 7.5 14.29 6.5 C14.29 5.42 13.47 4.6 12.4 4.6 C11.4 4.6 10.6 5.36 10.6 6.3 L8.6 6.2 C8.72 4.5 10.09 3.1 12.4 3.1 C14.84 3.1 16.3 4.5 16.3 6.42 C16.3 7.8 15.66 8.9 15.07 10.25 Z",
  spinner: "M12 2 C6.48 2 2 6.48 2 12 C2 17.52 6.48 22 12 22 C17.52 22 22 17.52 22 12 M12 2 C6.48 2 2 6.48 2 12",
  loading: "M12 2 C6.48 2 2 6.48 2 12",
  moreVertical: "M12 8 C13.1 8 14 7.1 14 6 C14 4.9 13.1 4 12 4 C10.9 4 10 4.9 10 6 C10 7.1 10.9 8 12 8 Z M12 14 C13.1 14 14 13.1 14 12 C14 10.9 13.1 10 12 10 C10.9 10 10 10.9 10 12 C10 13.1 10.9 14 12 14 Z M12 20 C13.1 20 14 19.1 14 18 C14 16.9 13.1 16 12 16 C10.9 16 10 16.9 10 18 C10 19.1 10.9 20 12 20 Z",
  moreHorizontal: "M6 10 C4.9 10 4 10.9 4 12 C4 13.1 4.9 14 6 14 C7.1 14 8 13.1 8 12 C8 10.9 7.1 10 6 10 Z M18 10 C16.9 10 16 10.9 16 12 C16 13.1 16.9 14 18 14 C19.1 14 20 13.1 20 12 C20 10.9 19.1 10 18 10 Z M12 10 C10.9 10 10 10.9 10 12 C10 13.1 10.9 14 12 14 C13.1 14 14 13.1 14 12 C14 10.9 13.1 10 12 10 Z",
  reload: "M7 7 C4.24 7 2 9.24 2 12 C2 14.56 3.68 16.73 6 17.73 L6 15.95 C4.29 15.16 3 13.32 3 11.13 C3 8.82 4.82 7 7 7 Z M17 7 C19.76 7 22 9.24 22 12 C22 14.13 20.71 15.97 18.7 16.76 L18.7 18.54 C21.32 17.54 23 15.56 23 13 C23 10.24 20.76 8 18 8 Z M11.5 5 L14.5 8 L8.5 8 Z M8.5 16 L14.5 16 L11.5 19 Z",
  arrowRight: "M5 12 L19 12 M13 6 L19 12 L13 18",
  arrowLeft: "M19 12 L5 12 M11 6 L5 12 L11 18",
  arrowUp: "M12 19 L12 5 M6 11 L12 5 L18 11",
  arrowDown: "M12 5 L12 19 M6 13 L12 19 L18 13",
  star: "M12 2 L15.09 10.26 L24 10.26 L17.55 15.13 L19.64 23.39 L12 18.53 L4.36 23.39 L6.45 15.13 L0 10.26 L8.91 10.26 Z",
  heart: "M20.84 4.61 C20.02 3.78 18.98 3.37 17.93 3.37 C16.88 3.37 15.84 3.78 15.02 4.61 L12 7.63 L8.98 4.61 C7.30 2.92 4.71 2.92 3.02 4.61 C1.33 6.30 1.33 8.88 3.02 10.57 L12 19.58 L20.99 10.57 C22.68 8.88 22.68 6.30 20.84 4.61 Z",
  palette: "M12 2 C6.48 2 2 6.48 2 12 C2 17.52 6.48 22 12 22 C17.52 22 22 17.52 22 12 C22 6.48 17.52 2 12 2 Z M7 12 C8.1 12 9 11.1 9 10 C9 8.9 8.1 8 7 8 C5.9 8 5 8.9 5 10 C5 11.1 5.9 12 7 12 Z M17 12 C18.1 12 19 11.1 19 10 C19 8.9 18.1 8 17 8 C15.9 8 15 8.9 15 10 C15 11.1 15.9 12 17 12 Z M12 17 C10.9 17 10 16.1 10 15 C10 13.9 10.9 13 12 13 C13.1 13 14 13.9 14 15 C14 16.1 13.1 17 12 17 Z M7 17 C5.9 17 5 16.1 5 15 C5 13.9 5.9 13 7 13 C8.1 13 9 13.9 9 15 C9 16.1 8.1 17 7 17 Z M17 17 C15.9 17 15 16.1 15 15 C15 13.9 15.9 13 17 13 C18.1 13 19 13.9 19 15 C19 16.1 18.1 17 17 17 Z",
  component: "M5 3 L19 3 L19 9 L5 9 Z M5 11 L19 11 L19 17 L5 17 Z M3 20 L21 20",
  table: "M3 4 L21 4 M3 4 L3 20 L21 20 L21 4 M3 8 L21 8 M3 12 L21 12 M3 16 L21 16 M8 4 L8 20 M13 4 L13 20 M18 4 L18 20",
  layout: "M3 3 L21 3 L21 21 L3 21 Z M3 8 L21 8 M3 3 L3 21 M8 8 L8 21",
  moon: "M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z",
  sun: "M12 3v2M12 19v2M5.64 5.64l1.42 1.42M16.95 16.95l1.41 1.41M3 12h2M19 12h2M5.64 18.36l1.41-1.41M16.95 7.05l1.41-1.41M12 7a5 5 0 1 0 0 10A5 5 0 0 0 12 7z"
}, B = ({ name: r, size: t = 24, className: s = "" }) => {
  const i = Ra[r];
  return i ? /* @__PURE__ */ a.jsx(
    "svg",
    {
      viewBox: "0 0 24 24",
      width: t,
      height: t,
      fill: "none",
      stroke: "currentColor",
      strokeWidth: "1.75",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: s,
      children: /* @__PURE__ */ a.jsx("path", { d: i })
    }
  ) : (console.warn(`Icon "${r}" does not exist in ICONS map`), null);
}, he = b.forwardRef(
  ({ variant: r = "primary", size: t = "md", className: s = "", disabled: i, ...o }, l) => {
    const d = [
      "btn",
      `btn--${r}`,
      `btn--${t}`,
      s
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "button",
      {
        ref: l,
        disabled: i,
        className: d,
        ...o
      }
    );
  }
);
he.displayName = "Button";
const wa = b.forwardRef(
  ({ variant: r = "neutral", form: t = "default", className: s = "", children: i, ...o }, l) => {
    const d = [
      "chip",
      `chip--${t}`,
      t === "default" && `chip--${r}`,
      s
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs(
      "span",
      {
        ref: l,
        className: d,
        ...o,
        children: [
          t === "default" && /* @__PURE__ */ a.jsx("span", { className: "chip__dot" }),
          t === "env" && /* @__PURE__ */ a.jsx("span", { className: "chip__dot chip__dot--env" }),
          i
        ]
      }
    );
  }
);
wa.displayName = "Chip";
const ka = b.forwardRef(
  ({ color: r = "cyan", pulse: t = !1, className: s = "", ...i }, o) => {
    const l = ["badge", `badge--${r}`, t && "badge--pulse", s].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "span",
      {
        ref: o,
        className: l,
        ...i
      }
    );
  }
);
ka.displayName = "Badge";
const Ta = b.forwardRef(
  ({ color: r = "cyan", pulse: t = !1, className: s = "", ...i }, o) => {
    const l = ["status-badge", `status-badge--${r}`, t && "status-badge--pulse", s].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "div",
      {
        ref: o,
        className: l,
        ...i
      }
    );
  }
);
Ta.displayName = "StatusBadge";
const Pa = b.forwardRef(
  ({ mono: r = !1, error: t = !1, className: s = "", ...i }, o) => {
    const l = [
      "text-input",
      r && "text-input--mono",
      t && "text-input--error",
      s
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "input",
      {
        ref: o,
        type: "text",
        className: l,
        ...i
      }
    );
  }
);
Pa.displayName = "TextInput";
const Sa = b.forwardRef(
  ({ mono: r = !1, error: t = !1, className: s = "", ...i }, o) => {
    const l = [
      "text-area",
      r && "text-area--mono",
      t && "text-area--error",
      s
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "textarea",
      {
        ref: o,
        className: l,
        ...i
      }
    );
  }
);
Sa.displayName = "TextArea";
const Da = b.forwardRef(
  ({ mono: r = !1, error: t = !1, className: s = "", ...i }, o) => {
    const l = [
      "number-input",
      r && "number-input--mono",
      t && "number-input--error",
      s
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "input",
      {
        ref: o,
        type: "number",
        className: l,
        ...i
      }
    );
  }
);
Da.displayName = "NumberInput";
const Oa = b.forwardRef(
  ({ error: r = !1, className: t = "", ...s }, i) => {
    const o = [
      "select",
      r && "select--error",
      t
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "select",
      {
        ref: i,
        className: o,
        ...s
      }
    );
  }
);
Oa.displayName = "Select";
const Za = b.forwardRef(
  ({ indeterminate: r = !1, className: t = "", ...s }, i) => {
    const o = b.useRef(null), l = b.useCallback(
      (m) => {
        o.current = m, typeof i == "function" ? i(m) : i && (i.current = m);
      },
      [i]
    );
    b.useEffect(() => {
      o.current && (o.current.indeterminate = r);
    }, [r]);
    const d = [
      "tri-state",
      r && "tri-state--indeterminate",
      t
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "input",
      {
        ref: l,
        type: "checkbox",
        className: d,
        ...s
      }
    );
  }
);
Za.displayName = "TriState";
const Ba = b.forwardRef(
  ({ label: r, required: t = !1, error: s, hint: i, children: o, className: l = "", ...d }, m) => {
    const u = ["field", l].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: m, className: u, ...d, children: [
      r && /* @__PURE__ */ a.jsxs("div", { className: "field__label", children: [
        /* @__PURE__ */ a.jsx("span", { children: r }),
        t && /* @__PURE__ */ a.jsx("span", { className: "field__required", children: "*" }),
        i && /* @__PURE__ */ a.jsx("span", { className: "field__hint", children: i })
      ] }),
      /* @__PURE__ */ a.jsx("div", { className: "field__input", children: o }),
      s && /* @__PURE__ */ a.jsx("div", { className: "field__error", children: s })
    ] });
  }
);
Ba.displayName = "Field";
const $a = b.forwardRef(
  ({ label: r, value: t, delta: s, color: i = "cyan", className: o = "", ...l }, d) => {
    const u = ["stat-tile", `stat-tile--${i}`, o].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: d, className: u, ...l, children: [
      /* @__PURE__ */ a.jsx("div", { className: "stat-tile__label", children: r }),
      /* @__PURE__ */ a.jsx("div", { className: "stat-tile__value", children: t }),
      s && /* @__PURE__ */ a.jsxs("div", { className: "stat-tile__meta", children: [
        /* @__PURE__ */ a.jsxs("span", { className: `stat-tile__delta stat-tile__delta--${s.direction || "up"}`, children: [
          s.direction === "down" ? "−" : "+",
          Math.abs(s.value)
        ] }),
        s.label && /* @__PURE__ */ a.jsx("span", { className: "stat-tile__label-secondary", children: s.label })
      ] })
    ] });
  }
);
$a.displayName = "StatTile";
const Ve = b.forwardRef(
  ({
    columns: r,
    data: t,
    rowKey: s,
    selectable: i = !0,
    selectedRows: o = [],
    onSelectRows: l,
    onSort: d,
    className: m = "",
    ...u
  }, p) => {
    const [C, v] = K(null), [j, f] = K("asc"), y = (L, S) => typeof s == "function" ? s(L, S) : L[s], k = () => {
      o.length === t.length ? l == null || l([]) : l == null || l(t.map((L, S) => y(L, S)));
    }, P = (L) => {
      o.includes(L) ? l == null || l(o.filter((S) => S !== L)) : l == null || l([...o, L]);
    }, M = (L) => {
      C === L ? j === "asc" ? (f("desc"), d == null || d(L, "desc")) : (v(null), d == null || d(L, null)) : (v(L), f("asc"), d == null || d(L, "asc"));
    }, E = ["table", m].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("table", { ref: p, className: E, ...u, children: [
      /* @__PURE__ */ a.jsx("thead", { className: "table__head", children: /* @__PURE__ */ a.jsxs("tr", { className: "table__row", children: [
        i && /* @__PURE__ */ a.jsx("th", { className: "table__header table__header--checkbox", style: { width: "30px" }, children: /* @__PURE__ */ a.jsx(
          "input",
          {
            type: "checkbox",
            className: "table__checkbox",
            checked: o.length === t.length && t.length > 0,
            onChange: k
          }
        ) }),
        r.map((L) => /* @__PURE__ */ a.jsx(
          "th",
          {
            className: `table__header ${L.sortable ? "table__header--sortable" : ""}`,
            style: { width: L.width },
            onClick: () => L.sortable && M(String(L.key)),
            children: /* @__PURE__ */ a.jsxs("div", { className: "table__header-content", children: [
              L.label,
              C === String(L.key) && j && /* @__PURE__ */ a.jsx(
                B,
                {
                  name: j === "asc" ? "chevronUp" : "chevronDown",
                  size: 14,
                  className: "table__sort-icon"
                }
              )
            ] })
          },
          String(L.key)
        ))
      ] }) }),
      /* @__PURE__ */ a.jsx("tbody", { className: "table__body", children: t.map((L, S) => {
        const O = y(L, S), A = o.includes(O);
        return /* @__PURE__ */ a.jsxs("tr", { "data-row-key": O, className: `table__row ${A ? "table__row--selected" : ""}`, children: [
          i && /* @__PURE__ */ a.jsx("td", { className: "table__cell table__cell--checkbox", children: /* @__PURE__ */ a.jsx(
            "input",
            {
              type: "checkbox",
              className: "table__checkbox",
              checked: A,
              onChange: () => P(O)
            }
          ) }),
          r.map(($) => /* @__PURE__ */ a.jsx("td", { className: "table__cell", children: $.render ? $.render(L[$.key], L, S) : L[$.key] }, `${O}-${String($.key)}`))
        ] }, O);
      }) })
    ] });
  }
);
Ve.displayName = "Table";
const er = Ve, Aa = b.forwardRef(
  ({ icon: r, label: t, count: s, active: i = !1, depth: o = 0, onClick: l, className: d = "", ...m }, u) => {
    const p = [
      "nav-item",
      i && "nav-item--active",
      o === 1 && "nav-item--depth-1",
      d
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs(
      "button",
      {
        ref: u,
        className: p,
        onClick: l,
        "aria-current": i ? "page" : void 0,
        ...m,
        children: [
          o === 0 && r && /* @__PURE__ */ a.jsx(B, { name: r, size: 16, className: "nav-item__icon" }),
          /* @__PURE__ */ a.jsx("span", { className: "nav-item__label", children: t }),
          s !== void 0 && /* @__PURE__ */ a.jsx("span", { className: "nav-item__count", children: s })
        ]
      }
    );
  }
);
Aa.displayName = "NavItem";
const Ke = b.forwardRef(
  ({
    sections: r,
    activeItemId: t,
    collapsed: s = !1,
    onCollapse: i,
    onSelectItem: o,
    className: l = "",
    ...d
  }, m) => {
    const [u, p] = K(/* @__PURE__ */ new Set()), C = (j) => {
      p((f) => {
        const y = new Set(f);
        return y.has(j) ? y.delete(j) : y.add(j), y;
      });
    }, v = [
      "sidebar",
      s && "sidebar--collapsed",
      l
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: m, className: v, ...d, children: [
      /* @__PURE__ */ a.jsx(
        "button",
        {
          className: "sidebar__toggle",
          onClick: () => i == null ? void 0 : i(!s),
          "aria-label": s ? "Expand sidebar" : "Collapse sidebar",
          children: /* @__PURE__ */ a.jsx(B, { name: "menu", size: 20 })
        }
      ),
      /* @__PURE__ */ a.jsx("nav", { className: "sidebar__nav", children: r.map((j) => /* @__PURE__ */ a.jsxs("div", { className: "sidebar__section", children: [
        !s && /* @__PURE__ */ a.jsx("div", { className: "sidebar__section-title", children: j.title }),
        /* @__PURE__ */ a.jsx("div", { className: "sidebar__items", children: j.items.map((f) => {
          const y = f.children && f.children.length > 0, k = u.has(f.id), P = t === f.id;
          return /* @__PURE__ */ a.jsxs(b.Fragment, { children: [
            /* @__PURE__ */ a.jsxs(
              "button",
              {
                className: `sidebar__item ${P ? "sidebar__item--active" : ""}`,
                onClick: () => {
                  y ? C(f.id) : o == null || o(f.id);
                },
                title: s ? f.label : void 0,
                children: [
                  f.icon && /* @__PURE__ */ a.jsx(B, { name: f.icon, size: 18, className: "sidebar__item-icon" }),
                  !s && /* @__PURE__ */ a.jsx("span", { className: "sidebar__item-label", children: f.label }),
                  !s && f.count !== void 0 && /* @__PURE__ */ a.jsx("span", { className: "sidebar__item-count", children: f.count }),
                  !s && y && /* @__PURE__ */ a.jsx(
                    B,
                    {
                      name: "chevronRight",
                      size: 14,
                      className: `sidebar__item-chevron ${k ? "sidebar__item-chevron--open" : ""}`
                    }
                  )
                ]
              }
            ),
            !s && y && k && f.children.map((M) => /* @__PURE__ */ a.jsxs(
              "button",
              {
                className: `sidebar__item sidebar__item--child ${t === M.id ? "sidebar__item--active" : ""}`,
                onClick: () => o == null ? void 0 : o(M.id),
                children: [
                  /* @__PURE__ */ a.jsx("span", { className: "sidebar__item-label", children: M.label }),
                  M.count !== void 0 && /* @__PURE__ */ a.jsx("span", { className: "sidebar__item-count", children: M.count })
                ]
              },
              M.id
            ))
          ] }, f.id);
        }) })
      ] }, j.title)) })
    ] });
  }
);
Ke.displayName = "Sidebar";
const He = b.forwardRef(
  ({
    breadcrumbs: r,
    searchPlaceholder: t = "Search...",
    onSearch: s,
    children: i,
    className: o = "",
    ...l
  }, d) => {
    const m = ["topbar", o].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: d, className: m, ...l, children: [
      /* @__PURE__ */ a.jsx("div", { className: "topbar__breadcrumbs", children: r && r.length > 0 && /* @__PURE__ */ a.jsx("nav", { className: "breadcrumbs", children: r.map((u, p) => /* @__PURE__ */ a.jsxs(b.Fragment, { children: [
        p > 0 && /* @__PURE__ */ a.jsx("span", { className: "breadcrumbs__separator", children: "/" }),
        u.href ? /* @__PURE__ */ a.jsx("a", { href: u.href, className: "breadcrumbs__link", children: u.label }) : /* @__PURE__ */ a.jsx(
          "button",
          {
            className: "breadcrumbs__link",
            onClick: u.onClick,
            style: { background: "none", border: "none", cursor: u.onClick ? "pointer" : "default" },
            children: u.label
          }
        )
      ] }, p)) }) }),
      /* @__PURE__ */ a.jsxs("div", { className: "topbar__actions", children: [
        s && /* @__PURE__ */ a.jsx(
          "input",
          {
            type: "text",
            placeholder: t,
            className: "topbar__search",
            onChange: (u) => s(u.target.value)
          }
        ),
        i
      ] })
    ] });
  }
);
He.displayName = "Topbar";
const Fa = b.forwardRef(
  ({ tabs: r, activeTabId: t, onSelectTab: s, className: i = "", ...o }, l) => {
    const d = ["tab-bar", i].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx("div", { ref: l, className: d, ...o, children: /* @__PURE__ */ a.jsx("div", { className: "tab-bar__tabs", children: r.map((m) => /* @__PURE__ */ a.jsxs(
      "button",
      {
        className: `tab-bar__tab ${t === m.id ? "tab-bar__tab--active" : ""}`,
        onClick: () => s(m.id),
        children: [
          /* @__PURE__ */ a.jsx("span", { className: "tab-bar__tab-label", children: m.label }),
          m.count !== void 0 && /* @__PURE__ */ a.jsx("span", { className: "tab-bar__tab-count", children: m.count })
        ]
      },
      m.id
    )) }) });
  }
);
Fa.displayName = "TabBar";
const ve = b.forwardRef(
  ({ title: r, version: t, collapsed: s = !1, className: i = "", ...o }, l) => {
    const d = [
      "app-title",
      s && "app-title--collapsed",
      i
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: l, className: d, ...o, children: [
      /* @__PURE__ */ a.jsx("div", { className: "app-title__mark" }),
      !s && /* @__PURE__ */ a.jsxs("div", { className: "app-title__text", children: [
        /* @__PURE__ */ a.jsx("div", { className: "app-title__name", children: r }),
        t && /* @__PURE__ */ a.jsx("span", { className: "app-title__version", children: t })
      ] })
    ] });
  }
);
ve.displayName = "AppTitle";
const Ia = b.forwardRef(
  ({ left: r, center: t, right: s, className: i = "", ...o }, l) => {
    const d = ["titlebar", i].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: l, className: d, ...o, children: [
      r && /* @__PURE__ */ a.jsx("div", { className: "titlebar__slot titlebar__slot--left", children: r }),
      t && /* @__PURE__ */ a.jsx("div", { className: "titlebar__slot titlebar__slot--center", children: t }),
      s && /* @__PURE__ */ a.jsx("div", { className: "titlebar__slot titlebar__slot--right", children: s })
    ] });
  }
);
Ia.displayName = "Titlebar";
const qe = b.forwardRef(
  ({ left: r, center: t, right: s, className: i = "", ...o }, l) => {
    const d = ["statusbar", i].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: l, className: d, ...o, children: [
      r && /* @__PURE__ */ a.jsx("div", { className: "statusbar__slot statusbar__slot--left", children: r }),
      t && /* @__PURE__ */ a.jsx("div", { className: "statusbar__slot statusbar__slot--center", children: t }),
      s && /* @__PURE__ */ a.jsx("div", { className: "statusbar__slot statusbar__slot--right", children: s })
    ] });
  }
);
qe.displayName = "Statusbar";
const za = b.forwardRef(
  ({
    appTitle: r,
    topbar: t,
    sidebar: s,
    statusbar: i,
    children: o,
    className: l = "",
    ...d
  }, m) => {
    const u = ["shell-layout", l].filter(Boolean).join(" "), { hide: p, ...C } = r ?? {}, v = r && !r.hide, { hide: j, ...f } = t ?? {}, y = t && !t.hide, { hide: k, ...P } = s ?? {}, M = s && !s.hide, { hide: E, ...L } = i ?? {}, S = i && !i.hide, O = P.collapsed ?? !1;
    return /* @__PURE__ */ a.jsxs("div", { ref: m, className: u, ...d, children: [
      /* @__PURE__ */ a.jsxs("div", { className: "shell-layout__main", children: [
        M ? /* @__PURE__ */ a.jsxs("div", { className: "shell-layout__sidebar-col", children: [
          v && /* @__PURE__ */ a.jsx(ve, { ...C, collapsed: O }),
          /* @__PURE__ */ a.jsx(Ke, { ...P })
        ] }) : v ? /* @__PURE__ */ a.jsx(ve, { ...C }) : null,
        /* @__PURE__ */ a.jsxs("div", { className: "shell-layout__content", children: [
          y && /* @__PURE__ */ a.jsx(He, { ...f }),
          /* @__PURE__ */ a.jsx("div", { className: "shell-layout__canvas", children: o })
        ] })
      ] }),
      S && /* @__PURE__ */ a.jsx(qe, { ...L })
    ] });
  }
);
za.displayName = "ShellLayout";
const Wa = [
  "a[href]",
  "button:not([disabled])",
  "textarea:not([disabled])",
  "input:not([disabled])",
  "select:not([disabled])",
  '[tabindex]:not([tabindex="-1"])'
];
function pe(r, t) {
  const s = W(null);
  Y(() => {
    if (!t || !r.current) return;
    s.current = document.activeElement;
    const i = r.current.querySelectorAll(
      Wa.join(",")
    ), o = (l) => {
      if (l.key !== "Tab" || i.length === 0) return;
      const d = i[0], m = i[i.length - 1];
      l.shiftKey ? document.activeElement === d && (l.preventDefault(), m.focus()) : document.activeElement === m && (l.preventDefault(), d.focus());
    };
    return i.length > 0 && (i[0].focus(), r.current.addEventListener("keydown", o)), () => {
      var l;
      (l = r.current) == null || l.removeEventListener("keydown", o), s.current && s.current.focus && s.current.focus();
    };
  }, [r, t]);
}
let re = 0, te = null;
function Ce(r) {
  Y(() => {
    if (r)
      return re === 0 && (te = document.body.style.overflow, document.body.style.overflow = "hidden"), re++, () => {
        re--, re === 0 && te !== null && (document.body.style.overflow = te, te = null);
      };
  }, [r]);
}
const Je = b.forwardRef(
  ({ isOpen: r, onClose: t, title: s, subtitle: i, children: o, footer: l, className: d = "", ...m }, u) => {
    const p = W(null), C = W(null);
    _e(u, () => C.current), pe(C, r), Ce(r), Y(() => {
      const j = (f) => {
        f.key === "Escape" && r && t();
      };
      if (r)
        return document.addEventListener("keydown", j), () => {
          document.removeEventListener("keydown", j);
        };
    }, [r, t]);
    const v = (j) => {
      j.target === p.current && t();
    };
    return r ? /* @__PURE__ */ a.jsx(
      "div",
      {
        ref: p,
        className: "modal-backdrop",
        onClick: v,
        children: /* @__PURE__ */ a.jsxs(
          "div",
          {
            ref: C,
            className: ["modal", d].filter(Boolean).join(" "),
            role: "dialog",
            "aria-modal": "true",
            ...m,
            children: [
              s && /* @__PURE__ */ a.jsxs("div", { className: "modal__header", children: [
                /* @__PURE__ */ a.jsxs("div", { className: "modal__header-text", children: [
                  /* @__PURE__ */ a.jsx("div", { className: "modal__title", children: s }),
                  i && /* @__PURE__ */ a.jsx("div", { className: "modal__subtitle", children: i })
                ] }),
                /* @__PURE__ */ a.jsx(
                  "button",
                  {
                    className: "modal__close",
                    onClick: t,
                    "aria-label": "Close dialog",
                    children: /* @__PURE__ */ a.jsx(B, { name: "x", size: 14 })
                  }
                )
              ] }),
              /* @__PURE__ */ a.jsx("div", { className: "modal__body", children: o }),
              l && /* @__PURE__ */ a.jsx("div", { className: "modal__footer", children: l })
            ]
          }
        )
      }
    ) : null;
  }
);
Je.displayName = "Modal";
const Ya = b.forwardRef(
  ({
    isOpen: r,
    onClose: t,
    onConfirm: s,
    title: i,
    message: o,
    confirmLabel: l = "Confirm",
    cancelLabel: d = "Cancel",
    variant: m = "danger"
  }, u) => {
    const p = () => {
      s(), t();
    };
    return /* @__PURE__ */ a.jsxs(
      Je,
      {
        ref: u,
        isOpen: r,
        onClose: t,
        title: i,
        children: [
          /* @__PURE__ */ a.jsx("div", { className: "confirm-dialog__message", children: o }),
          /* @__PURE__ */ a.jsxs("div", { className: "confirm-dialog__footer", children: [
            /* @__PURE__ */ a.jsx(
              he,
              {
                variant: "ghost",
                size: "sm",
                onClick: t,
                children: d
              }
            ),
            /* @__PURE__ */ a.jsx(
              he,
              {
                variant: m,
                size: "sm",
                onClick: p,
                children: l
              }
            )
          ] })
        ]
      }
    );
  }
);
Ya.displayName = "ConfirmDialog";
const Ua = {
  success: "check",
  error: "x",
  warning: "alert",
  info: "info"
}, Va = {
  success: "emerald",
  error: "rose",
  warning: "amber",
  info: "cyan"
}, Ka = b.forwardRef(
  ({
    isOpen: r,
    onClose: t,
    title: s,
    subtitle: i,
    variant: o = "info",
    icon: l,
    duration: d = 4e3,
    className: m = "",
    ...u
  }, p) => {
    if (Y(() => {
      if (r && d) {
        const j = setTimeout(t, d);
        return () => clearTimeout(j);
      }
    }, [r, d, t]), !r) return null;
    const C = l || Ua[o], v = Va[o];
    return /* @__PURE__ */ a.jsxs(
      "div",
      {
        ref: p,
        className: ["toast", `toast--${o}`, m].filter(Boolean).join(" "),
        role: "status",
        "aria-live": "polite",
        ...u,
        children: [
          /* @__PURE__ */ a.jsx("div", { className: `toast__icon toast__icon--${v}`, children: /* @__PURE__ */ a.jsx(B, { name: C, size: 12 }) }),
          /* @__PURE__ */ a.jsxs("div", { className: "toast__content", children: [
            /* @__PURE__ */ a.jsx("div", { className: "toast__title", children: s }),
            i && /* @__PURE__ */ a.jsx("div", { className: "toast__subtitle", children: i })
          ] }),
          /* @__PURE__ */ a.jsx(
            "button",
            {
              className: "toast__close",
              onClick: t,
              "aria-label": "Dismiss notification",
              children: /* @__PURE__ */ a.jsx(B, { name: "x", size: 14 })
            }
          )
        ]
      }
    );
  }
);
Ka.displayName = "Toast";
const Ha = b.forwardRef(
  ({ isOpen: r, onClose: t, commands: s, placeholder: i = "Search commands..." }, o) => {
    const [l, d] = K(""), [m, u] = K(0), p = W(null), C = W(null);
    _e(o, () => C.current), pe(C, r), Ce(r);
    const v = s.filter(
      (f) => {
        var y;
        return f.label.toLowerCase().includes(l.toLowerCase()) || (((y = f.description) == null ? void 0 : y.toLowerCase().includes(l.toLowerCase())) ?? !1);
      }
    );
    Y(() => {
      r ? setTimeout(() => {
        var f;
        return (f = p.current) == null ? void 0 : f.focus();
      }, 0) : (d(""), u(0));
    }, [r]), Y(() => {
      const f = (y) => {
        r && (y.key === "Escape" ? t() : y.key === "ArrowDown" ? (y.preventDefault(), v.length > 0 && u((k) => (k + 1) % v.length)) : y.key === "ArrowUp" ? (y.preventDefault(), v.length > 0 && u((k) => (k - 1 + v.length) % v.length)) : y.key === "Enter" && (y.preventDefault(), v[m] && (v[m].onSelect(), t(), d(""))));
      };
      return document.addEventListener("keydown", f), () => document.removeEventListener("keydown", f);
    }, [r, t, v, m]);
    const j = (f) => {
      f.target === f.currentTarget && t();
    };
    return r ? /* @__PURE__ */ a.jsx(
      "div",
      {
        ref: C,
        className: "command-palette-backdrop",
        onClick: j,
        children: /* @__PURE__ */ a.jsxs("div", { className: "command-palette", children: [
          /* @__PURE__ */ a.jsxs("div", { className: "command-palette__header", children: [
            /* @__PURE__ */ a.jsx(B, { name: "search", size: 16 }),
            /* @__PURE__ */ a.jsx(
              "input",
              {
                ref: p,
                type: "text",
                className: "command-palette__input",
                placeholder: i,
                value: l,
                onChange: (f) => {
                  d(f.target.value), u(0);
                }
              }
            )
          ] }),
          /* @__PURE__ */ a.jsx("div", { className: "command-palette__list", children: v.length === 0 ? /* @__PURE__ */ a.jsx("div", { className: "command-palette__empty", children: "No commands found" }) : v.map((f, y) => /* @__PURE__ */ a.jsxs(
            "button",
            {
              className: ["command-palette__item", y === m && "command-palette__item--selected"].filter(Boolean).join(" "),
              onClick: () => {
                f.onSelect(), t(), d("");
              },
              children: [
                f.icon && /* @__PURE__ */ a.jsx(B, { name: f.icon, size: 16 }),
                /* @__PURE__ */ a.jsxs("div", { className: "command-palette__item-content", children: [
                  /* @__PURE__ */ a.jsx("div", { className: "command-palette__item-label", children: f.label }),
                  f.description && /* @__PURE__ */ a.jsx("div", { className: "command-palette__item-description", children: f.description })
                ] })
              ]
            },
            f.id
          )) })
        ] })
      }
    ) : null;
  }
);
Ha.displayName = "CommandPalette";
const qa = b.forwardRef(
  ({ title: r, subtitle: t, footer: s, bordered: i = !0, className: o = "", children: l, ...d }, m) => {
    const u = ["panel", !i && "panel--no-border", o].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs("div", { ref: m, className: u, ...d, children: [
      (r || t) && /* @__PURE__ */ a.jsx("div", { className: "panel__header", children: /* @__PURE__ */ a.jsxs("div", { children: [
        r && /* @__PURE__ */ a.jsx("div", { className: "panel__title", children: r }),
        t && /* @__PURE__ */ a.jsx("div", { className: "panel__subtitle", children: t })
      ] }) }),
      /* @__PURE__ */ a.jsx("div", { className: "panel__body", children: l }),
      s && /* @__PURE__ */ a.jsx("div", { className: "panel__footer", children: s })
    ] });
  }
);
qa.displayName = "Panel";
const Ja = b.forwardRef(
  ({ isOpen: r, onClose: t, title: s, position: i = "right", width: o = "320px", children: l, className: d = "", ...m }, u) => {
    const p = W(null);
    _e(u, () => p.current), pe(p, r), Ce(r), Y(() => {
      const v = (j) => {
        j.key === "Escape" && r && t();
      };
      if (r)
        return document.addEventListener("keydown", v), () => {
          document.removeEventListener("keydown", v);
        };
    }, [r, t]);
    const C = (v) => {
      v.target === v.currentTarget && t();
    };
    return r ? /* @__PURE__ */ a.jsx(
      "div",
      {
        className: "drawer-backdrop",
        onClick: C,
        children: /* @__PURE__ */ a.jsxs(
          "div",
          {
            ref: p,
            className: ["drawer", `drawer--${i}`, d].filter(Boolean).join(" "),
            style: { width: o },
            role: "dialog",
            "aria-modal": "true",
            ...m,
            children: [
              s && /* @__PURE__ */ a.jsxs("div", { className: "drawer__header", children: [
                /* @__PURE__ */ a.jsx("h2", { className: "drawer__title", children: s }),
                /* @__PURE__ */ a.jsx(
                  "button",
                  {
                    className: "drawer__close",
                    onClick: t,
                    "aria-label": "Close drawer",
                    children: /* @__PURE__ */ a.jsx(B, { name: "x", size: 14 })
                  }
                )
              ] }),
              /* @__PURE__ */ a.jsx("div", { className: "drawer__body", children: l })
            ]
          }
        )
      }
    ) : null;
  }
);
Ja.displayName = "Drawer";
const Ga = b.forwardRef(
  ({
    direction: r = "horizontal",
    initialSplitPercent: t = 50,
    minSize: s = 200,
    maxSize: i = 800,
    first: o,
    second: l,
    className: d = "",
    ...m
  }, u) => {
    const [p, C] = K(t), v = W(null), j = (k) => (P) => {
      k.forEach((M) => {
        typeof M == "function" ? M(P) : M && (M.current = P);
      });
    }, f = () => {
      const k = (M) => {
        if (!v.current) return;
        const E = v.current.getBoundingClientRect(), L = r === "horizontal", S = L ? M.clientX - E.left : M.clientY - E.top, O = L ? E.width : E.height;
        let A = S / O * 100;
        const $ = s / O * 100, ne = i / O * 100;
        A = Math.max($, Math.min(A, ne)), C(A);
      }, P = () => {
        document.removeEventListener("mousemove", k), document.removeEventListener("mouseup", P);
      };
      document.addEventListener("mousemove", k), document.addEventListener("mouseup", P);
    }, y = ["split-pane", `split-pane--${r}`, d].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsxs(
      "div",
      {
        ref: j([v, u]),
        className: y,
        ...m,
        children: [
          /* @__PURE__ */ a.jsx(
            "div",
            {
              className: "split-pane__first",
              style: r === "horizontal" ? { width: `${p}%` } : { height: `${p}%` },
              children: o
            }
          ),
          /* @__PURE__ */ a.jsx(
            "div",
            {
              className: `split-pane__divider split-pane__divider--${r}`,
              onMouseDown: f
            }
          ),
          /* @__PURE__ */ a.jsx(
            "div",
            {
              className: "split-pane__second",
              style: r === "horizontal" ? { width: `${100 - p}%` } : { height: `${100 - p}%` },
              children: l
            }
          )
        ]
      }
    );
  }
);
Ga.displayName = "SplitPane";
const Xa = b.forwardRef(
  ({ columns: r = 4, className: t = "", children: s, ...i }, o) => {
    const l = ["stat-grid", `stat-grid--cols-${r}`, t].filter(Boolean).join(" ");
    return /* @__PURE__ */ a.jsx(
      "div",
      {
        ref: o,
        className: l,
        ...i,
        children: s
      }
    );
  }
);
Xa.displayName = "StatGrid";
export {
  ve as AppTitle,
  ka as Badge,
  he as Button,
  wa as Chip,
  Ha as CommandPalette,
  Ya as ConfirmDialog,
  Ja as Drawer,
  Ba as Field,
  B as Icon,
  Je as Modal,
  Aa as NavItem,
  Da as NumberInput,
  qa as Panel,
  Oa as Select,
  za as ShellLayout,
  Ke as Sidebar,
  Ga as SplitPane,
  Xa as StatGrid,
  $a as StatTile,
  Ta as StatusBadge,
  qe as Statusbar,
  Fa as TabBar,
  er as Table,
  Sa as TextArea,
  Pa as TextInput,
  Ia as Titlebar,
  Ka as Toast,
  He as Topbar,
  Za as TriState
};
