// @tinkermonkey/heimdall-ui/src/App.tsx
import { useState, useMemo } from 'react'
import {
  Button,
  Chip,
  Badge,
  StatusBadge,
  StatTile,
  Table,
  ShellLayout,
  TabBar,
} from './index'
import type { Column } from './components/Table'
import ContextStudioRebuilt from './examples/ContextStudioRebuilt'
import HomelabDashboardRebuilt from './examples/HomelabDashboardRebuilt'
import PrimitivesTestPage from './test-pages/PrimitivesTestPage'
import DataDisplayTestPage from './test-pages/DataDisplayTestPage'
import ShellFrameworkTestPage from './test-pages/ShellFrameworkTestPage'
import OverlayComponentsTestPage from './test-pages/OverlayComponentsTestPage'
import AdvancedOverlayComponentsTestPage from './test-pages/AdvancedOverlayComponentsTestPage'
import FoundationTestPage from './test-pages/FoundationTestPage'
import NavigationComponentTestPage from './test-pages/NavigationComponentTestPage'

export default function App() {
  const [selectedRows, setSelectedRows] = useState<(string | number)[]>([])
  const [activeTab, setActiveTab] = useState('overview')
  const [sidebarCollapsed, setSidebarCollapsed] = useState(false)

  const exampleParam = useMemo(() => {
    const params = new URLSearchParams(window.location.search)
    return params.get('example')
  }, [])

  const testParam = useMemo(() => {
    const params = new URLSearchParams(window.location.search)
    return params.get('test')
  }, [])

  const tableData = [
    { id: 'cls_4f3a', name: 'organism', class: 'life', status: 'active', updated: '2m ago' },
    { id: 'cls_8b21', name: 'station', class: 'climate', status: 'syncing', updated: '12m ago' },
    { id: 'cls_e007', name: 'service', class: 'software', status: 'error', updated: '1h ago' },
  ]

  const tableColumns: Column<typeof tableData[number]>[] = [
    {
      key: 'id',
      label: 'ID',
      width: '120px',
      render: (value: string) => <span style={{ fontFamily: 'var(--font-mono)', fontSize: '12px' }}>{value}</span>,
    },
    { key: 'name', label: 'Name' },
    { key: 'class', label: 'Class' },
    { key: 'status', label: 'Status' },
    { key: 'updated', label: 'Updated', render: (value: string) => <span style={{ fontFamily: 'var(--font-mono)', fontSize: '12px' }}>{value}</span> },
  ]

  if (exampleParam === 'rebuilt') {
    return <ContextStudioRebuilt />
  }

  if (exampleParam === 'homelab') {
    return <HomelabDashboardRebuilt />
  }

  if (exampleParam === 'primitives') {
    return <PrimitivesTestPage />
  }

  if (exampleParam === 'data-display') {
    return <DataDisplayTestPage />
  }

  if (exampleParam === 'shell-framework') {
    return <ShellFrameworkTestPage />
  }

  if (exampleParam === 'overlays') {
    return <OverlayComponentsTestPage />
  }

  if (exampleParam === 'advanced-overlays') {
    return <AdvancedOverlayComponentsTestPage />
  }

  if (exampleParam === 'navigation') {
    return <NavigationComponentTestPage />
  }

  if (testParam === 'foundation') {
    return <FoundationTestPage />
  }

  return (
    <ShellLayout
      appTitle={{ title: 'Heimdall', version: 'v0.1.0' }}
      topbar={{
        breadcrumbs: [
          { label: 'Dashboard' },
          { label: 'Overview' },
        ],
        searchPlaceholder: 'Search entities...',
      }}
      sidebar={{
        collapsed: sidebarCollapsed,
        onCollapse: setSidebarCollapsed,
        sections: [
          {
            title: 'Workspace',
            items: [
              { id: 'dashboard', label: 'Dashboard', icon: 'dashboard' },
              { id: 'schema', label: 'Schema', icon: 'schema', count: 128 },
              { id: 'individuals', label: 'Individuals', icon: 'data', count: 12480 },
              { id: 'pipelines', label: 'Pipelines', icon: 'pipeline', count: 17 },
            ],
          },
        ],
        activeItemId: 'schema',
      }}
      statusbar={{
        right: <span>UI · Phase 6 · Shell Framework</span>,
      }}
    >
      <div style={{ maxWidth: '1200px' }}>
        <h1 style={{ fontSize: '2.25rem', fontWeight: '800', marginBottom: '1rem', color: 'rgb(var(--canvas-fg-1))' }}>
          Heimdall Design System
        </h1>
        <p style={{ color: 'rgb(var(--canvas-fg-2))', marginBottom: '2rem' }}>Phase 6: Shell Framework</p>

        {/* StatTile Section */}
        <section style={{ marginBottom: '2rem' }}>
          <h2 style={{ fontSize: '1.5rem', fontWeight: '700', marginBottom: '1rem', color: 'rgb(var(--canvas-fg-1))' }}>
            StatTile
          </h2>
          <div style={{ display: 'grid', gridTemplateColumns: 'repeat(4, 1fr)', gap: '14px', marginBottom: '2rem' }}>
            <StatTile label="Classes" value="128" color="cyan" delta={{ value: 4, label: 'this week', direction: 'up' }} />
            <StatTile label="Individuals" value="12,480" color="violet" delta={{ value: 312, label: 'today', direction: 'up' }} />
            <StatTile label="Pipelines" value="17" color="amber" />
            <StatTile label="Uptime" value="99.8%" color="emerald" delta={{ value: 0.1, label: '24h', direction: 'down' }} />
          </div>
        </section>

        {/* TabBar Section */}
        <section style={{ marginBottom: '2rem' }}>
          <h2 style={{ fontSize: '1.5rem', fontWeight: '700', marginBottom: '1rem', color: 'rgb(var(--canvas-fg-1))' }}>
            TabBar
          </h2>
          <TabBar
            tabs={[
              { id: 'overview', label: 'Overview', count: 3 },
              { id: 'details', label: 'Details', count: 12 },
              { id: 'activity', label: 'Activity' },
            ]}
            activeTabId={activeTab}
            onSelectTab={setActiveTab}
          />
        </section>

        {/* Table Section */}
        <section style={{ marginBottom: '2rem' }}>
          <h2 style={{ fontSize: '1.5rem', fontWeight: '700', marginBottom: '1rem', color: 'rgb(var(--canvas-fg-1))' }}>
            Table
          </h2>
          <Table
            columns={tableColumns}
            data={tableData}
            rowKey="id"
            selectedRows={selectedRows}
            onSelectRows={setSelectedRows}
          />
        </section>

        {/* Primitives Section */}
        <section style={{ marginBottom: '2rem' }}>
          <h2 style={{ fontSize: '1.5rem', fontWeight: '700', marginBottom: '1rem', color: 'rgb(var(--canvas-fg-1))' }}>
            Primitive Components
          </h2>
          <div style={{ marginBottom: '2rem' }}>
            <h3 style={{ fontSize: '1.125rem', fontWeight: '600', marginBottom: '0.75rem', color: 'rgb(var(--canvas-fg-1))' }}>
              Buttons
            </h3>
            <div style={{ display: 'flex', gap: '0.75rem', flexWrap: 'wrap' }}>
              <Button variant="primary">Primary</Button>
              <Button variant="secondary">Secondary</Button>
              <Button variant="ghost">Ghost</Button>
              <Button variant="danger">Delete</Button>
            </div>
          </div>

          <div style={{ marginBottom: '2rem' }}>
            <h3 style={{ fontSize: '1.125rem', fontWeight: '600', marginBottom: '0.75rem', color: 'rgb(var(--canvas-fg-1))' }}>
              Chips
            </h3>
            <div style={{ display: 'flex', gap: '0.5rem', flexWrap: 'wrap' }}>
              <Chip variant="cyan">cyan</Chip>
              <Chip variant="amber">amber</Chip>
              <Chip variant="violet">violet</Chip>
              <Chip variant="emerald">emerald</Chip>
            </div>
          </div>

          <div>
            <h3 style={{ fontSize: '1.125rem', fontWeight: '600', marginBottom: '0.75rem', color: 'rgb(var(--canvas-fg-1))' }}>
              Badges
            </h3>
            <div style={{ display: 'flex', gap: '2rem', flexWrap: 'wrap' }}>
              <Badge color="cyan" />
              <Badge color="emerald" />
              <Badge color="amber" />
              <StatusBadge color="cyan" pulse />
            </div>
          </div>
        </section>
      </div>
    </ShellLayout>
  )
}
